plugins/scripting/scripting-ide-services/src/org/jetbrains/kotlin/scripting/ide_services/compiler/KJvmReplCompilerWithIdeServices.kt

```diff
@@ -48,13 +48,18 @@ class KJvmReplCompilerWithIdeServices(hostConfiguration: ScriptingHostConfigurat
                 }
             }
 
+            val filterOutShadowed = configuration[ScriptCompilationConfiguration.completion.filterOutShadowedDescriptors]!!
+            val nameFilter = configuration[ScriptCompilationConfiguration.completion.nameFilter]!!
+
             with(analyzeResult.valueOr { return it }) {
                 return getKJvmCompletion(
                     ktScript,
                     bindingContext,
                     resolutionFacade,
                     moduleDescriptor,
-                    cursorAbs
+                    cursorAbs,
+                    filterOutShadowed,
```

Comment:
Why don't you pass the configuration right into the function and extract values from it right in place, inside `KJvmReplCompleter`? It will look much cleaner to me and will require fewer changes when you'll add more options to it.

Reply:
Tried to choose between "flat" structure and less files to change. Your arguments seem reasonable, I'll fix it.